backends/magma/ceed-magma-gemm-selector.cpp

```diff
@@ -92,57 +80,50 @@ void gemm_selector(int gpu_arch, char precision, char transA, int m, int n, int
   }
 
   if (ir >= 0) {
-    *use_magma = (*data)[ir][USE_MAGMA_INDEX];
     // if the closest match indicates that n = nbatch,
     // that means calling the regular non-batch gemm.
     // So nbatch is set to n instead of the 'nbatch'
     // entry of the matching record
-    int n_      = (*data)[ir][N_INDEX];
-    int nbatch_ = (*data)[ir][N_BATCH_INDEX];
+    int n_      = data[ir][N_INDEX];
+    int nbatch_ = data[ir][N_BATCH_INDEX];
     *nbatch     = (n_ == nbatch_) ? n : nbatch_;
+    *use_magma  = data[ir][USE_MAGMA_INDEX];
   }
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-static inline void *nontensor_rtc_get_data(int gpu_arch, char precision, CeedEvalMode emode, CeedTransposeMode tmode) {
-#ifdef CEED_MAGMA_USE_HIP
-  void *data = (void *)&dinterp_n_mi250x;
-#else
-  void *data = (void *)&dinterp_n_a100;
-#endif
-
 #ifdef CEED_MAGMA_USE_HIP
-  if (emode == CEED_EVAL_INTERP) {
-    data = (tmode == CEED_TRANSPOSE) ? (void *)&dinterp_t_mi250x : (void *)&dinterp_n_mi250x;
+static inline auto nontensor_rtc_get_data(int gpu_arch, char transA, int qcomp) -> decltype(dinterp_n_mi250x) {
+  if (qcomp == 1) {
+    return (transA == 'n') ? dinterp_n_mi250x : dinterp_t_mi250x;
   } else {
-    data = (tmode == CEED_TRANSPOSE) ? (void *)&dgrad_t_mi250x : (void *)&dgrad_n_mi250x;
+    return (transA == 'n') ? dgrad_n_mi250x : dgrad_t_mi250x;
   }
+}
 #else
-  if (emode == CEED_EVAL_INTERP) {
-    data = (tmode == CEED_TRANSPOSE) ? (void *)&dinterp_t_a100 : (void *)&dinterp_n_a100;
+static inline auto nontensor_rtc_get_data(int gpu_arch, char transA, int qcomp) -> decltype(dinterp_n_a100) {
+  if (qcomp == 1) {
+    return (transA == 'n') ? dinterp_n_a100 : dinterp_t_a100;
   } else {
-    data = (tmode == CEED_TRANSPOSE) ? (void *)&dgrad_t_a100 : (void *)&dgrad_n_a100;
+    return (transA == 'n') ? dgrad_n_a100 : dgrad_t_a100;
   }
-#endif
-
-  return data;
 }
+#endif
 
 ////////////////////////////////////////////////////////////////////////////////
-CeedInt nontensor_rtc_get_nb(int gpu_arch, char precision, CeedEvalMode emode, CeedTransposeMode tmode, int P_, int N, int Q_) {
-  CeedInt P  = (tmode == CEED_TRANSPOSE) ? P_ : Q_;
-  CeedInt Q  = (tmode == CEED_TRANSPOSE) ? Q_ : P_;
+CeedInt nontensor_rtc_get_nb(int gpu_arch, char transA, int qcomp, int P_, int N, int Q_) {
+  CeedInt P  = (transA == 'n') ? P_ : Q_;
+  CeedInt Q  = (transA == 'n') ? Q_ : P_;
```

Comment:
Welp, this is certainly a difference from the previous behavior. Let me fix that.

Reply:
Fixed this one in [511d170](https://github.com/CEED/libCEED/pull/1301/commits/511d170bb98bfe27abfe974249cd74d63808d730).